File: output/gemma4-e4b/avatar/Python/Java/codeforces_274_A.java
import java.util.*;
import java.io.IOException;
import java.util.stream.Collectors;

public class codeforces_274_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read num and k from the first line
        if (!scanner.hasNextInt()) {
            return;
        }
        int num = scanner.nextInt();
        int k = scanner.nextInt();

        // Read all remaining integers into the array list (arr)
        List<Integer> arrList = new ArrayList<>();
        while (scanner.hasNextInt()) {
            arrList.add(scanner.nextInt());
        }
        
        scanner.close();

        // Sort the array
        Collections.sort(arrList);

        // Initialization
        Set<Integer> different = new HashSet<>();
        int temp = 0;

        // Iterate through the sorted elements
        for (int x : arrList) {
            // Python condition: (x % k != 0 or not x // k in different)
            // In Java, integer division (x / k) performs floor division for positive numbers.
            boolean conditionMet = (x % k != 0) || !different.contains(x / k);
            
            if (conditionMet) {
                different.add(x);
            }
            
            // temp = max(len(different), temp)
            temp = Math.max(different.size(), temp);
        }

        System.out.println(temp);
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_274_A.py
import sys

def solve():
    # Read all input tokens
    data = sys.stdin.read().split()
    
    if not data:
        return

    # n is the first element, k is the second
    n = int(data[0])
    k = int(data[1])
    
    # arr elements start from index 2
    arr = [int(x) for x in data[2:n+2]]
    
    arr.sort()
    
    count = 0
    s = set()
    
    for x in arr:
        if x % k != 0:
            count += 1
            s.add(x)
        else:
            # x % k == 0
            if (x // k) not in s:
                count += 1
                s.add(x)
                
    print(count)

solve()